geoutils/raster/raster.py

```diff
@@ -3192,23 +3373,81 @@ def xy2ij(
 
         return i, j
 
-    def ij2xy(self, i: ArrayLike, j: ArrayLike, offset: str = "ul") -> tuple[NDArrayNum, NDArrayNum]:
+    def ij2xy(
+        self, i: ArrayLike, j: ArrayLike, shift_area_or_point: bool | None = None, force_offset: str | None = None
+    ) -> tuple[NDArrayNum, NDArrayNum]:
         """
         Get coordinates (x,y) of indexes (row,column).
 
-        Defaults to upper-left, for which this function is fully reversible with xy2ij.
+        By default, the indexes are shifted with the interpretation of pixel coordinates "AREA_OR_POINT" of the
+        raster, to ensure that the indexes of points represent the right location. See parameter description of
+        shift_area_or_point for more details.
+
+        This function is reversible with xy2ij for any pixel interpretation.
 
         :param i: Row (i) index of pixel.
         :param j: Column (j) index of pixel.
-        :param offset: Return coordinates as "center" of pixel, or any corner (upper-left "ul", "ur", "ll", lr").
+        :param shift_area_or_point: Whether to shift with pixel interpretation, which shifts to center of pixel
+            coordinates if self.area_or_point is "Point" and maintains corner pixel coordinate if it is "Area" or None.
+            Defaults to True. Can be configured with the global setting geoutils.config["shift_area_or_point"].
+        :param force_offset: Ignore pixel interpretation and force coordinate to a certain offset: "center" of pixel, or
+            any corner (upper-left "ul", "ur", "ll", lr"). Default coordinate of a raster is upper-left.
 
         :returns x, y: x,y coordinates of i,j in reference system.
         """
 
-        x, y = rio.transform.xy(self.transform, i, j, offset=offset)
+        # If undefined, default to the global system config
+        if shift_area_or_point is None:
+            shift_area_or_point = config["shift_area_or_point"]
+
+        # Shift by half a pixel back for "Point" interpretation
+        if shift_area_or_point and force_offset is None:
+            if self.area_or_point is not None and self.area_or_point == "Point":
+                i = np.asarray(i) - 0.5
+                j = np.asarray(j) - 0.5
```

Comment:
Why is the sign different from the shift applied at line 845?

Reply:
That is the last (and quite important) thing that I need to check (last box in the to-do list of the PR). I was wondering about the best way to check this, I could just rely on GDAL's documentation... Or GDAL's printed extents depending on the metadata as shown in https://github.com/GlacioHack/geoutils/issues/503 (first link)

Reply:
Right now it's probably wrong

Reply:
It was actually right!
The shift in `set_area_or_point` shifts coordinates, so needs the negative sign of the Y, hence the different sign. While in `xy2ij`, it's only an index with (0, 0) starting at the upper left corner.
I added a new suite of test to check this against #503, all fully consistent and passing!

Reply:
See [4cf8899](https://github.com/GlacioHack/geoutils/pull/509/commits/4cf8899f717c119d0f511c104a48f7260292534a) for changes